crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs

```diff
@@ -202,10 +315,63 @@ impl ValidationDependencyState {
         }
     }
 
+    pub(super) fn set_pending_validation_warranted(
+        &mut self,
+        action: SignedActionHashed,
+        op_type: ChainOpType,
+    ) {
+        if self.dependency_type == ValidationDependencyType::Action {
+            tracing::warn!("Attempted to set an action on a dependency that is not of type Action, this is a bug");
+            return;
+        }
+
+        match self.dependency {
+            None => {
+                self.dependency = Some(ValidationDependency {
+                    value: ValidationDependencyValue::Warranted(WarrantedDep::Pending(
+                        action, op_type,
+                    )),
+                    fetched_from: CascadeSource::Network,
+                });
+            }
+            _ => {
+                tracing::warn!("Attempted to set a record on a dependency that already has a value, this is a bug")
+            }
+        }
+    }
+
+    pub(super) fn set_validation_status_for_warranted(
+        &mut self,
+        validation_status: ValidationStatus,
+    ) {
+        if self.dependency_type == ValidationDependencyType::Action {
+            tracing::warn!("Attempted to set an action on a dependency that is not of type Action, this is a bug");
+            return;
+        }
+
+        match self.dependency.take() {
```

Comment:
Why are we using `.take()` here? It seems to me like there are cases other than `None` that would be covered by the second match arm below that only logs a warning but in that case the existing value would still be overwritten by `None`. Is this intentional?

Reply:
Because the inner action of the old value needs to be borrowed while also mutating the top level of it. The compiler rightly doesn't let you do that. 

There shouldn't be cases where this takes but is unable to modify, that is supposed to be prevented by the guard condition at the start of the function

Reply:
That's true. I think quite a bit would have to go wrong to hit that though and it will print a warning. Worst case scenario is that the data has to be fetched from the database again but if the state logic is broken then things aren't going to progress either way.

Reply:
Okay, fair enough :+1: